packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -6,75 +6,108 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {DOCUMENT, ɵgetDOM as getDOM} from '@angular/common';
+import {DOCUMENT} from '@angular/common';
 import {Inject, Injectable, OnDestroy} from '@angular/core';
 
 @Injectable()
-export class SharedStylesHost {
-  /** @internal */
-  protected _stylesSet = new Set<string>();
+export class SharedStylesHost implements OnDestroy {
+  private readonly usedStyles = new Map<string /** Style string */, number /** Usage count */>();
```

Comment:
nit: may be we can rename this a bit to convey its role:
```suggestion
  private readonly usageCount = new Map<string /** Style string */, number /** Usage count */>();
```